File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        my_dict1 = dict()
        my_dict2 = dict()

        for i in s:
            if i in my_dict1:
                my_dict1[i] = my_dict1.get(i) + 1
            else:
                my_dict1.update({i:1})
        for i in t:
            if i in my_dict2:
                my_dict2[i] = my_dict2.get(i) + 1
            else:
                my_dict2.update({i:1})

        if my_dict1 == my_dict2:
            return True
        else:
            return False

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
    public:
        bool isAnagram(string s, string t) {
            
            if (s.size() != t.size()) {
                return false;
            }

            int check_array[256] = {0};
            for (int i = 0; i < s.size(); i++) {
                check_array[(int)s[i]]++;
                check_array[(int)t[i]]--;

            }            
            for (int i = 0; i < 256; i++) {
                if (check_array[i] != 0) {
                    return false;
                }
            }
            return true;
        }
    };